sending-a-simple-message-to-a-service/README.md

```diff
@@ -0,0 +1,325 @@
+
+```ballerina
+log:printInfo("Retrieving information", specialization = category);
+
+Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+if resp is Doctor[] {
+    return resp;
+}
+
+log:printError("Retrieving doctor information failed", resp);
+if resp is http:ClientRequestError {
+    return <http:NotFound> {body: string `category not found: ${category}`};
+}
+
+return <http:InternalServerError> {body: resp.message()};
+```
+
+#### Complete source
+
+You have successfully developed the required service.
+
+```ballerina
+import ballerina/http;
+import ballerina/log;
+
+type Doctor record {|
+    string name;
+    string hospital;
+    string category;
+    string availability;
+    decimal fee;
+|};
+
+configurable int port = 8290;
+configurable string healthcareBackend = "http://localhost:9090/healthcare";
+
+final http:Client queryDoctorEP = check initializeHttpClient();
+
+function initializeHttpClient() returns http:Client|error => new (healthcareBackend);
+
+service /healthcare on new http:Listener(port) {
+    resource function get querydoctor/[string category]() 
+            returns Doctor[]|http:NotFound|http:InternalServerError {
+        log:printInfo("Retrieving information", specialization = category);
+        
+        Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+        if resp is Doctor[] {
+            return resp;
+        }
+        
+        log:printError("Retrieving doctor information failed", resp);
+        if resp is http:ClientRequestError {
+            return <http:NotFound> {body: string `category not found: ${category}`};
+        }
+
+        return <http:InternalServerError> {body: resp.message()};
+    }
+}
+```
+
+#### Diagram
+
+The [sequence diagram view](https://wso2.com/ballerina/vscode/docs/implement-the-code/sequence-diagram-view/) for the implemented resource method is the following.
+
+![sequence diagram](./resources/tutorial_sending_a_simple_message_to_a_service.png)
+
+### Step 3: Build and run the service
+
+You can run this service by navigating to the project root and using the `bal run` command.
+
+```bash
+sending-a-simple-message-to-a-service$ bal run
+Compiling source
+        integration_tutorials/sending_a_simple_message_to_a_service:0.1.0
+
+Running executable
+```
+
+### Step 4: Try out the use case
+
+Let's test the use case by sending a request to the service.
+
+#### Start the back end service
+
+Download the JAR file for the backend service from [here](https://github.com/ballerina-guides/integration-tutorials/blob/main/backends/hospital-service/) and execute the following command to start the service:
+
+```bash
+bal run hospitalservice.jar
+```
+
+#### Send a request
+
+Let's send a request to the service using cURL as follows.
+
+1. Install and set up [cURL](https://curl.se/) as your client.
```

Comment:
```suggestion
1. Install and set up [cURL](https://curl.se/) as your client.
```
Do we need to mention this? If so, should we add it under a main `Prerequisites` section?

Reply:
It's not a prerequisite. Just one way to do it.